src/charm.py

```diff
@@ -239,7 +244,7 @@ def _validate(self):
             raise ValueError("peer relation not ready")
 
         # Validate config.
-        valid_services = ("frontend", "history", "matching", "worker")
+        valid_services = ("frontend", "history", "matching", "worker", "internal-frontend")
```

Comment:
If you are going to make heavy use of these I wonder if it would make sense to convert these to an enum. I see that below we use `frontend` as a literal and I think that at some point passing around magic values becomes too unwieldy.

A conversion here can enable easy use of `match` statements as well, so combining the logic for `frontend` and `internal-frontend` would not be too bad.